cmd/geth/stakecmd.go

```diff
@@ -1157,16 +1157,19 @@ func setCommissionRate(ctx *cli.Context) error {
 		utils.Fatalf("Failed to MAX_VALID_COMMISSION_RATE: %v", err)
 	}
 
-	if rate.Cmp(big.NewInt(0)) != 0 && (minRate.Cmp(rate) > 0 || maxRate.Cmp(rate) < 0) {
-		utils.Fatalf("Commission rate should be 0 or between %.2f and %.2f", params.ToRayFloat64(minRate), params.ToRayFloat64(maxRate))
+	absRate := new(big.Int).Abs(rate)
+	if absRate.Cmp(big.NewInt(0)) != 0 && (minRate.Cmp(absRate) > 0 || maxRate.Cmp(absRate) < 0) {
+		utils.Fatalf("Commission rate should be 0 or between ±%.2f and ± %.2f", params.ToRayFloat64(minRate), params.ToRayFloat64(maxRate))
 	}
 
-	// send transaction
-
-	log.Info("Set commission rate", "rootchain", rootchainAddr, "commissionRate", params.ToRayFloat64(rate))
-
 	isCommissionRateNegative := rate.Cmp(big.NewInt(0)) < 0
-	absRate := new(big.Int).Abs(rate)
+	_commission := params.ToRayFloat64(rate)
```

Comment:
fix https://github.com/Onther-Tech/plasma-evm/blob/fix/stakecmd-negInput/params/denomination.go#L32-L57 functions to support the sign of `big.Int` and `float64`